ovirtbmc/ovirtbmc.py

```diff
@@ -43,29 +44,33 @@ def __init__(self, authdata, port, address, vm_name, cache_status, engine_fqdn,
         )
         self.vms_service = self.ovirt_conn.system_service().vms_service()
         self.vm = None
+        self.vm_name = None
         self.cache_disabled = not cache_status
         self.cached_status = None
         self.target_status = None
         # At times the bmc service is started before important things like networking have fully initialized. Keep
         # trying to find the vm indefinitely, since there's no point in continuing if we don't have an vm.
         while True:
             try:
-                self.vm = self._find_vm(vm_name)
-                if self.vm is not None:
-                    self.log(f'Managing vm: {vm_name} ID: {self.vm}')
-                    break
+                self.vm = self._find_vm(vm)
+                self.vm_name = self.vms_service.service(self.vm).get().name
+                self.log(f'Managing vm: {self.vm_name} ID: {self.vm}')
+                break
             except Exception as e:
-                self.log(f'Exception finding vm "{vm_name}": {e}')
+                self.log(f'Exception finding vm "{vm}": {e}')
                 time.sleep(1)
 
-    def _find_vm(self, vm_name):
-        vms = (vm for vm in self.vms_service.list() if vm.name == vm_name)
+    def _find_vm(self, vm):
         try:
-            vm_id = next(vms).id
-            return vm_id
-        except StopIteration:
-            self.log(f'Could not find specified vm {vm_name}')
-            sys.exit(1)
+            self.vms_service.service(vm).get()
```

Comment:
Doesn't this return a `VmBase` instance? In line 70 we're returning the `id` directly

Reply:
If this code works, then `vm` should be id and we can return it directly. Adjusted the code to make it clearer